File: src/0287-Find-the-Duplicate-Number/find_the_duplicate_number.py
from typing import List, Set

class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        nums_set: Set[int] = set()
        for num in nums:
            if num in nums_set:
                return num
            
            nums_set.add(num)

if __name__ == "__main__":
    nums: List[int] = [1,3,4,2,2]
    solution = Solution()
    
    print(solution.findDuplicate(nums))

File: src/0287-Find-the-Duplicate-Number/find_the_duplicate_number.cpp
#include <iostream>
#include <unordered_set>

class Solution {
public:
    int findDuplicate(std::vector<int>& nums) {
        std::unordered_set<int> nums_set;
        for (int num : nums) {
            if (nums_set.find(num) != nums_set.end()) {
                return num;
            }
            nums_set.insert(num);
        }
        return -1;
    }
};

int main() {
    std::vector<int> nums = {1, 3, 4, 2, 2};
    Solution solution;
    
    std::cout << solution.findDuplicate(nums);

    return 0;
}